sled-agent/src/nexus.rs

```diff
@@ -18,50 +18,54 @@ use std::sync::Arc;
 use tokio::sync::mpsc;
 use tokio::task::JoinHandle;
 
-struct Inner {
-    log: Logger,
-    resolver: Resolver,
-}
-
-/// Wrapper around a [`NexusClient`] object, which allows deferring
-/// the DNS lookup until accessed.
-///
-/// Without the assistance of OS-level DNS lookups, the [`NexusClient`]
-/// interface requires knowledge of the target service IP address.
-/// For some services, like Nexus, this can be painful, as the IP address
-/// may not have even been allocated when the Sled Agent starts.
+/// A thin wrapper over a progenitor-generated NexusClient.
 ///
-/// This structure allows clients to access the client on-demand, performing
-/// the DNS lookup only once it is actually needed.
+/// Also attaches the "DNS resolver" for historical reasons.
 #[derive(Clone)]
-pub struct LazyNexusClient {
-    inner: Arc<Inner>,
+pub struct NexusClientWithResolver {
+    client: NexusClient,
+    resolver: Arc<Resolver>,
 }
 
-impl LazyNexusClient {
-    pub fn new(log: Logger, addr: Ipv6Addr) -> Result<Self, ResolveError> {
+impl NexusClientWithResolver {
+    pub fn new(
```

Comment:
I wonder if we could name this and/or document it more clearly.  It's weird that this is "a thin wrapper" over the progenitor-generated NexusClient but its constructor arguments are so different, and it's surprising too that it would take a sled agent address.

Why not have the client be responsible for creating the Resolver so that we don't leak the policy choice ("the DNS resolvers are these fixed addresses relative to the sled IP") into this abstraction?

Reply:
I'd be happy with that choice -- updated to pass the `Resolver` as an argument. Agreed that it's better to not smear that policy decision across API boundaries.